extensions/positron-r/src/session.ts

```diff
@@ -806,8 +806,9 @@ export class RSession implements positron.LanguageRuntimeSession, vscode.Disposa
 	 * Resolves to `true` once the LSP is connected. Resolves to `false` if the
 	 * LSP has been stopped. Rejects if the LSP fails to start.
 	 */
-	async waitLsp(): Promise<boolean> {
-		return await this._lsp.wait();
+	async waitLsp(): Promise<ArkLsp> {
+		await this._lsp.wait();
+		return this._lsp;
```

Comment:
Check out the comment here, this actually used to represent 3 states:
- Connected
- Stopped
- Failed to connect

At the very least we should update this comment because it is out of date now, but did you consider these three states being reduced to 2 now?

Reply:
Good catch! This is now:

```ts
	 * Resolves to `ArkLsp` if the LSP is connected, or once the LSP is connected
	 * if it's starting up. Resolves to `undefined` if the LSP has been stopped. Rejects
	 * if the LSP fails to start.
```